samples/bluetooth/iso_broadcast/src/main.c

```diff
@@ -197,14 +212,87 @@ int main(void)
 			net_buf_reserve(buf, BT_ISO_CHAN_SEND_RESERVE);
 			sys_put_le32(iso_send_count, iso_data);
 			net_buf_add_mem(buf, iso_data, sizeof(iso_data));
-			ret = bt_iso_chan_send(&bis_iso_chan[chan], buf, seq_num);
+
+			/* Ensure that first ISO data on all streams are given with sequence number
+			 * 0, so that the (Zephyr) Controller uses the same payload counter for
+			 * a given sequence number on all streams at an SDU interval.
+			 * Read Tx Sync interface returns error if no ISO data was given previously.
+			 */
+			ret = bt_iso_chan_get_tx_sync(&bis_iso_chan[chan], &tx_info);
+			if (ret) {
```

Comment:
This may not work with all controllers. 

First reason:
The HCI command specifies 
> The Packet_Sequence_Number parameter contains the sequence number of a transmitted SDU.
At the time of calling `bt_iso_chan_get_tx_sync` here, we do not know if the SDU was transmitted. We may need to wait for the `sent` callback for that. 

Second reason:
Even in Core Spec 6.0 it still states 
> Host can use the HCI_LE_Read_ISO_TX_Sync command at any time to read the time stamp and packet sequence number of the last SDU **scheduled for transmission**.
in 4.7 LE Host and Controller synchronization. We have seen controllers return the PSN and timestamp of a transmitted SDU, and we have seen controllers return the PSN and timestamp of a scheduled transmission. 

I'm not sure how to best handle either of this complications

Reply:
I will go by Bluetooth Core Specification v6.0, Vol 4, Part E, Section 7.8.96 LE Read ISO TX Sync command

```
If the Host issues this command before an SDU has been transmitted by the Controller,
the Controller shall return the error code Command Disallowed (0x0C).
```

Regarding Bluetooth Core Specification v6.0, Vol 4, Part E, Section 4.7 LE Host and Controller synchronization

```
A Host can use the HCI_LE_Read_ISO_TX_Sync command at any time to read the
time stamp and packet sequence number of the last SDU scheduled for transmission.
```

I will go by `last SDU scheduled for transmission` because the Host which is not aware of the Controller clock will not know correctly when to call LE Read ISO TX Sync command such that `an SDU has been transmitted`.

As stated in the HCI Part of the specification, a conformant HCI Controller shall return `Command Disallowed (0x0C)` for LE Read ISO TX Sync command for the purpose used in this sample.